### PR TITLE
Fix three issues related to signal PDFs

### PR DESCRIPTION
**Pull request by dvandyk**
- refactors the ``SignalPDFs`` code to reflect what we have been doing for ``Observables`` (#610)
- add list of known signal PDFs to the documentation (#733)
- handle non-existing PDFs gracefully in the factory method (#748)

### CODE REVIEW

File: eos/Makefile.am [hunk 1]
```diff
@@ -73,7 +74,8 @@ LDADD = \
 check_PROGRAMS = \
 	constraint_TEST \
 	observable_TEST \
-	reference_TEST
+	reference_TEST \
+	signal-pdf_TEST
```
Comment (mreboud) [inline eos/Makefile.am:78]: Unrelated to this PR: Why does this list appear twice?

File: eos/b-decays/signal-pdfs.cc [hunk 2]
```diff
@@ -0,0 +1,453 @@
...
+                make_signal_pdf("B->gammalnu::d^2Gamma/dEgamma/dcos(theta_l)",
+                    Options{ },
+                    &BToGammaLeptonNeutrino::fully_differential_decay_width,
+                    std::make_tuple(
+                        KinematicRange{ "Egamma", 0.1, 2.64, BToGammaLeptonNeutrino::kinematics_description_Egamma },
+                        KinematicRange{ "cos(theta_l)", -1.0, +1.0, BToGammaLeptonNeutrino::kinematics_description_c_theta_l}
+                    ),
+                    &BToGammaLeptonNeutrino::integrated_branching_ratio,
+                    std::make_tuple(
+                        "E_gamma_min"
```
Comment (mreboud) [inline eos/b-decays/signal-pdfs.cc:54]: Can we use the same convention `E_gamma` or `Egamma` throughout EOS?

File: eos/signal-pdf.cc [hunk 3]
```diff
@@ -19,22 +19,13 @@
...
-#include <eos/rare-b-decays/b-to-kstar-ll.hh>
-#include <eos/rare-b-decays/b-to-psd-nu-nu.hh>
-#include <eos/rare-b-decays/b-to-vec-nu-nu.hh>
+#include <eos/b-decays/signal-pdfs.hh>
+#include <eos/rare-b-decays/signal-pdfs.hh>
 #include <eos/utils/density.hh>
+#include <eos/utils/instantiation_policy-impl.hh>
+#include <eos/utils/log.hh>
 #include <eos/utils/private_implementation_pattern-impl.hh>
 #include <eos/utils/wrapped_forward_iterator-impl.hh>
```
Comment (mreboud) [inline eos/signal-pdf.cc:31]: Can you remove lines 55 - 56 below? In the function `norm`, the `using std::pow` is not needed.

File: python/eos/signal_pdf.py [hunk 4]
```diff
@@ -167,11 +167,88 @@ def filter_entry(self, qn):
...
+                group_entries = 0
+                group_result  = fr'''
+                    <tbody>
+                        <tr>
+                            <th style="text-align:left" colspan=1>
+                                <a style="text-decoration: none" onclick="toggle_group(this, 'grp{group_id}')">{group.name()}</a>
+                            </th>
+                        </tr>
+                    </tbody>
+                '''#.format(group=group.name(), id=group_id)
```
Comment (mreboud) [inline python/eos/signal_pdf.py:231]: This comment might be removed, maybe?

**Review by mreboud — APPROVED** [review]
I am happy with the changes, I haven't check that the documentation looked nice, but I guess you did. Thanks for the work!

File: eos/Makefile.am (hunk 1 above)
Comment (dvandyk) [reply eos/Makefile.am:78]: Ah, a peculiarity of automake. It has two targets: ``check-PROGRAMS`` and ``check-TESTS``. The former build the list of all programs (``check_PROGRAMS``) that must be available to the tests. The latter run all the tests in ``TESTS``.

File: eos/signal-pdf.cc (hunk 3 above)
Comment (dvandyk) [reply eos/signal-pdf.cc:31]: Done.

File: python/eos/signal_pdf.py (hunk 4 above)
Comment (dvandyk) [reply python/eos/signal_pdf.py:231]: Done.

File: eos/b-decays/signal-pdfs.cc (hunk 2 above)
Comment (dvandyk) [reply eos/b-decays/signal-pdfs.cc:54]: Yes. That should be done more systematically though. I have opened #750 and will resolve this conversation.

**Pull request merged**